From 284b94558379852464a67e8c72bed0ec27e80cdd Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Thu, 14 Aug 2025 12:17:37 +0100
Subject: [PATCH] keycloak: default FIPS mode to on

Default to FIPS feature.
Default to FIPS mode strict.

This will enable automatic ignore of bcprov jars, and will enable
loading symlinked bc-fips jars.
---
 common/src/main/java/org/keycloak/common/Profile.java       | 2 +-
 .../src/main/java/org/keycloak/config/SecurityOptions.java  | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/common/src/main/java/org/keycloak/common/Profile.java b/common/src/main/java/org/keycloak/common/Profile.java
index 2c9edbe708..efa561e151 100755
--- a/common/src/main/java/org/keycloak/common/Profile.java
+++ b/common/src/main/java/org/keycloak/common/Profile.java
@@ -98,7 +98,7 @@ public class Profile {
 
         UPDATE_EMAIL("Update Email Action", Type.PREVIEW),
 
-        FIPS("FIPS 140-2 mode", Type.DISABLED_BY_DEFAULT),
+        FIPS("FIPS 140-2 mode", Type.DEFAULT),
 
         DPOP("OAuth 2.0 Demonstrating Proof-of-Possession at the Application Layer", Type.PREVIEW),
 
diff --git a/quarkus/config-api/src/main/java/org/keycloak/config/SecurityOptions.java b/quarkus/config-api/src/main/java/org/keycloak/config/SecurityOptions.java
index 236dfd74b6..dae163badb 100644
--- a/quarkus/config-api/src/main/java/org/keycloak/config/SecurityOptions.java
+++ b/quarkus/config-api/src/main/java/org/keycloak/config/SecurityOptions.java
@@ -13,9 +13,9 @@ public class SecurityOptions {
             .expectedValues(getFipsModeValues())
             .buildTime(true)
             .description("Sets the FIPS mode. If '" + FipsMode.NON_STRICT + "' is set, FIPS is enabled but on non-approved mode. For full FIPS compliance, set '" + FipsMode.STRICT + "' to run on approved mode. "
-                    + "This option defaults to '" + FipsMode.DISABLED + "' when '" + Profile.Feature.FIPS.getKey() + "' feature is disabled, which is by default. "
-                    + "This option defaults to '" + FipsMode.NON_STRICT + "' when '" + Profile.Feature.FIPS.getKey() + "' feature is enabled.")
-            .defaultValue(FipsMode.DISABLED)
+                    + "This option defaults to '" + FipsMode.DISABLED + "' when '" + Profile.Feature.FIPS.getKey() + "' feature is disabled. "
+                    + "This option defaults to '" + FipsMode.STRICT + "' when '" + Profile.Feature.FIPS.getKey() + "' feature is enabled, which is by default.")
+            .defaultValue(FipsMode.STRICT)
             .build();
 
     private static List<String> getFipsModeValues() {
-- 
2.48.1

